cinemy/cinema/signals/handlers.py

```diff
@@ -0,0 +1,11 @@
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from cinema.models import generate_seats, Seat, Hall
+
+
+@receiver(post_save, sender=Hall)
+def hall_creation_handler(instance, **kwargs):
+    generated_seats = generate_seats(instance.rows, instance.seats_per_row)
+    for i in range(len(generated_seats)):
+        generated_seats[i].hall = instance
```

Comment:
Not 100% sure, but see if this might help, instead of the hall assignment for each object:
https://docs.djangoproject.com/en/4.0/ref/models/relations/#django.db.models.fields.related.RelatedManager.add

Reply:
Done